2226/2/versus.rb

```diff
@@ -0,0 +1,197 @@
+      end
+    end
+    swear_words_storage.size
+  end
+
+  def average_number_swearing_words_in_battle(name)
+    average = number_of_swear_words(name).fdiv(number_of_battles(name))
+    average.round(2)
+  end
+
+  def number_of_words_in_rounds(name)
+    number_of_words_in_rounds = 0
+
+    Dir.glob('rap-battles/*') do |file_name|
+      if file_name.match(name)
+        text = File.read(file_name)
+        number_of_words_in_rounds += text.split(' ').length
+      end
+    end
+    number_of_words_in_rounds
+  end
+
+  def number_of_rounds(name)
+    round_words = []
+
+    Dir.glob('rap-battles/*') do |file_name|
+      if file_name.match(name)
+        text = File.read(file_name)
+        round_words << text.match(/Раунд/)
+        round_words.reject!(&:nil?)
+      end
+    end
+    round_words
+  end
+
+  def average_number_words_in_round(name)
+    number_of_rounds = number_of_rounds(name).size
+    number_of_rounds = 1 if number_of_rounds.zero?
+
+    number_of_words_in_rounds(name) / number_of_rounds
+  end
+
+  def sort_and_reverse_hash(hash_for_sorting, sorted_storage)
+    hash_for_sorting = hash_for_sorting.sort_by { |_key, value| value[1] }
+    hash_for_sorting.reverse!
+    hash_for_sorting.each do |elem|
+      sorted_storage[elem[0]] = elem[1]
+    end
+    sorted_storage
+  end
+
+  def statistics(number)
+    unsorted_storage = {}
+    sorted_storage = {}
+    list_all_rapers.each do |raper|
+      storage_for_each_raper = []
+      storage_for_each_raper << number_of_battles(raper) << number_of_swear_words(raper)
+      storage_for_each_raper << average_number_swearing_words_in_battle(raper) << average_number_words_in_round(raper)
+      unsorted_storage[raper] = storage_for_each_raper
+    end
+    sorted_storage = sort_and_reverse_hash(unsorted_storage, sorted_storage)
+    sorted_storage.first(number.to_i)
+  end
+
+  # rubocop:disable Metrics/AbcSize
+  def make_table(number)
+    filtered_data = {}
+    statistics(number).each do |elem|
+      filtered_data[elem[0]] = elem[1]
+    end
+    rows = []
+    filtered_data.each do |k, v|
+      rows << [k.to_s, v[0].to_s + ' баттлов', v[1].to_s + ' нецензурных слов',
+               v[2].to_s + ' слова на баттл', v[3].to_s + ' слова в раунде']
+    end
+    Terminal::Table.new rows: rows
+  end
+  # rubocop:enable Metrics/AbcSize
+
+  def count_all_duplicates(array_storage)
+    array_storage.each_with_object(Hash.new(0)) { |word, counts| counts[word] += 1 }
+  end
+
+  def storage_for_all_text(name)
+    text_storage = ''
+    if list_all_rapers.include? name
+      Dir.glob('rap-battles/*') do |file_name|
+        if file_name.match(name)
+          text = File.read(file_name)
+          text_storage += text
+        end
+      end
+    end
+    text_storage
+  end
+
+  def the_most_used_words(top_words, name)
+    array_storage = storage_for_all_text(name).gsub(/,|'/, ' ').split.reject! { |x| x.length < 4 }
+    array_storage.map(&:capitalize!)
```

Comment:
better use `map!` and `capitalize`(without bang) it's easier to mutate one object than thousand